### PR TITLE
Scheduled Commands

### PR DESCRIPTION
**Pull request by adriangmweb**
## Description
Add the scheduled command decorator in order to be able to schedule tasks to run at a specific time or periodically.

## Checks
- [x] Project Builds
- [x] Project passes tests and checks
- [x] Updated documentation accordingly
 
## Additional information
If no parameters are provided to the decorator, the task will not run.

### CODE REVIEW

File: packages/framework-integration-tests/src/scheduled-commands/CheckCartCount.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { ScheduledCommand } from '@boostercloud/framework-core'
+import { Register, UUID } from '@boostercloud/framework-types'
+import { ProductCreated } from '../events/ProductCreated'
+
+@ScheduledCommand({
+  minute: '0/5',
+})
+export class CheckCartCount {
+  public static async handle(register: Register): Promise<void> {
+    console.log('This is a ScheduledCommand running', register)
```
Comment (juanjoman) [inline packages/framework-integration-tests/src/scheduled-commands/CheckCartCount.ts:10]: Maybe this one too? Or does it make sense to keep it?

File: packages/framework-provider-aws/src/library/scheduled-adapter.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { ScheduledCommandEnvelope, UUID } from '@boostercloud/framework-types'
+import { Logger } from '@boostercloud/framework-types'
```
Comment (juanjoman) [inline packages/framework-provider-aws/src/library/scheduled-adapter.ts:2]:
```suggestion
import { Logger, ScheduledCommandEnvelope, UUID } from '@boostercloud/framework-types'
```

**Review by juanjoman — APPROVED** [review]
LGTM!! Just left a couple of minor suggestions 🎉

File: packages/framework-integration-tests/integration/end-to-end/cart.integration.ts [hunk 3]
```diff
@@ -435,4 +435,36 @@ describe('Cart end-to-end tests', () => {
...
+                }
+              }
+            `,
+          })
+        },
+        (result) => result?.data?.ProductReadModels?.some((product: any) => product.sku === scheduledProductSku)
+      )
+
+      const product = products.data.ProductReadModels.find((product: any) => product.sku === scheduledProductSku)
+      console.log({ product })
```
[the remark this answers is not in the document]
Comment (adriangmweb) [reply packages/framework-integration-tests/integration/end-to-end/cart.integration.ts:466]: thanks, man! Totally forgot about it 😂

File: packages/framework-integration-tests/integration/end-to-end/cart.integration.ts [hunk 4]
```diff
@@ -435,4 +435,35 @@ describe('Cart end-to-end tests', () => {
...
+                }
+              }
+            `,
+          })
+        },
+        (result) => result?.data?.ProductReadModels?.some((product: any) => product.sku === scheduledProductSku)
+      )
+
+      const product = products.data.ProductReadModels.find((product: any) => product.sku === scheduledProductSku)
+      expect(product).not.to.be.null
```
Comment (javiertoledo) [inline packages/framework-integration-tests/integration/end-to-end/cart.integration.ts:466]: Great test!

**Review by javiertoledo — CHANGES_REQUESTED** [review]
Great code, the only thing I'm missing is a note in the documentation about this new feature and basic usage, could you please add it?

File: packages/framework-integration-tests/integration/end-to-end/cart.integration.ts (hunk 4 above)
Comment (adriangmweb) [reply packages/framework-integration-tests/integration/end-to-end/cart.integration.ts:466]: Thanks, @javiertoledo! You can find some notes about it in the [README](https://github.com/boostercloud/booster/blob/a10f305a7502a7cfd7ac367db5cb31a420f063b9/docs/README.md#scheduling-a-command)

**Review by NickSeagull — APPROVED** [review]

File: docs/README.md [hunk 5]
```diff
@@ -1027,15 +1029,41 @@ where the schema for `CreateProductInput` is
...
+```
+
+Notice that you can pass as parameters `minute`, `hour`, `day`, `month`, `weekDay` and `year` to set up a cron expression. By default, if no paramateres are passed, the scheduled command will not be triggered.
+
+#### Creating a scheduled command
+
+The preferred way to create a scheduled command is by using the generator, e.g.
+
+```shell
+boost new:scheduled-command CheckCartCount
```
Comment (javiertoledo) [inline docs/README.md:1056]: One nice thing we could do in this generator is giving the user a chance to define the cron schedule right in the CLI like this:

```shell
boost new:scheduled-command CheckCartCount --minutes "0/5"
```

Anyway, let's merge this as it is today and do this in a separate issue, this PR is long enough!

File: docs/README.md [hunk 6]
```diff
@@ -1027,15 +1029,41 @@ where the schema for `CreateProductInput` is
...
+})
+export class CheckCartCount {
+  public static async handle(): Promise<void> {
+    /* YOUR CODE HERE */
+  }
+}
+```
+
+Notice that you can pass as parameters `minute`, `hour`, `day`, `month`, `weekDay` and `year` to set up a cron expression. By default, if no paramateres are passed, the scheduled command will not be triggered.
+
```
Comment (alvaroloes) [inline docs/README.md:1050]: ❤️

**Review by javiertoledo — APPROVED** [review]
Awesome, 🚢🇮🇹❗

File: docs/README.md [hunk 7]
```diff
@@ -1568,44 +1600,50 @@ authorize them to access protected resources.
...
+| ---------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| _clientId_       | The application client Id that you got as an output when the application was deployed.                                                                                                               |
+| _username_       | The username of the user you want to register. It **must be an email**.                                                                                                                              |
+| _password_       | The password the user will use to later login into your application and get access tokens.                                                                                                           |
+| _userAttributes_ | Here you can specify the attributes of your user. These are: <br/> -_**role**_: A unique role this user will have. You can only specify here a role where the `signUpOptions` property is not empty. |
```
Comment (alvaroloes) [inline docs/README.md:1639]: What happened with all these changes in format? Does this affect the rendered Markdown?

File: docs/README.md [hunk 8]
```diff
@@ -2018,17 +2098,15 @@ socket.
...
 ```
+
 ```graphql
 mutation {
-  ChangeCart(input: {
-    cartId: "demo"
-    sku: "ABC_02"
-    quantity: 3
-  })
+  ChangeCart(input: { cartId: "demo", sku: "ABC_02", quantity: 3 })
```
Comment (alvaroloes) [inline docs/README.md:2104]: I don't think these changes improve readability...

File: packages/cli/src/commands/new/scheduled-command.ts [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+import { templates } from '../../templates'
+import { checkItIsABoosterProject } from '../../services/project-checker'
+
+export default class ScheduledCommand extends Oclif.Command {
+  public static description = "generate new scheduled command, write 'boost new:scheduled-command -h' to see options"
+  public static flags = {
+    help: Oclif.flags.help({ char: 'h' }),
+    fields: Oclif.flags.string({
+      char: 'f',
+      description: 'fields that this scheduled command will contain',
```
Comment (alvaroloes) [inline packages/cli/src/commands/new/scheduled-command.ts:16]: A scheduled command doesn't have fields. What is this flag used for?

File: packages/framework-provider-aws-infrastructure/src/infrastructure/stacks/scheduled-commands-stack.ts [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+export class ScheduledCommandStack {
+  public constructor(
+    private readonly config: BoosterConfig,
+    private readonly stack: Stack,
+    private readonly apis: APIs
+  ) {}
+
+  public build(): ScheduledCommandStackMembers {
+    const scheduledLambda = this.buildLambda('scheduled-task', this.config.scheduledTaskHandler)
```
Comment (alvaroloes) [inline packages/framework-provider-aws-infrastructure/src/infrastructure/stacks/scheduled-commands-stack.ts:20]: I would add a check here to see if there are any scheduled commands defined by the user. If not, it is not needed to create this lambda. 

We do the same thing in the AuthStack: We don't create anything if there are no roles defined.

**Review by alvaroloes — COMMENTED** [review]
GREAT PR! I left some comments. The following two are the most important ones that I think you should consider:
- 
-

File: docs/README.md (hunk 8 above)
Comment (adriangmweb) [reply docs/README.md:2104]: It was the format on save linter 🤦

**Comment by NickSeagull** [comment]
Please can someone explain why is it `minutes: 0/5` instead of `minutes: 5`? 😭 

@adriangmweb

**Comment by adriangmweb** [comment]
> Please can someone explain why is it `minutes: 0/5` instead of `minutes: 5`? 😭
> 
> @adriangmweb

Sure, it's a weird format

Using just `minutes: 5` will result in `run at minute 5 of every hour` like [in here](https://crontab.guru/#5_*_*_*_*)

While using `minutes: 0/5` results in `run every 5 minutes` like [in here](https://crontab.guru/#0/5_*_*_*_*)

**Pull request merged**